python/segyio/utils.py

```diff
@@ -146,6 +146,25 @@ def __repr__(self):
         return msg.format(self.name, self.byte, self.type, self.requires_nonzero_value)
 
 
+class TraceHeaderLayout:
+    def __init__(self, entries):
+        self.entries = list(entries)
+        self.by_byte = {entry.byte: entry for entry in self.entries}
+        self.by_name = {entry.name: entry for entry in self.entries}
+
+    def get_by_byte(self, byte):
+        return self.by_byte.get(byte)
+
+    def get_by_name(self, name):
+        return self.by_name.get(name)
```

Comment:
Depending on how we want to access/work with the members of the class, we could drop the `get_` methods and directly access the members similar to the approach in `TraceHeaderLayoutEntry`.